### ISSUE TITLE
Enable setting batch size in `add_columns`

### ISSUE THREAD

**Issue by tonyf** [lance-format/lance#2801]
Default batch size for lance may cause OOM depending on the size of a row and available system memory

**Comment by westonpace** [comment]
Yes, we should make this configurable.  However, in the meantime, you can use `LANCE_DEFAULT_BATCH_SIZE` for spots where we don't make it configurable.

**Comment by tonyf** [comment]
Is `num_threads` configurable via an environment variable as well?

**Comment by tonyf** [comment]
nvm, looks like it is via `LANCE_IO_THREADS`

**Comment by westonpace** [comment]
There is also a `LANCE_CPU_THREADS`.  The difference between `LANCE_IO_THREADS` and `LANCE_CPU_THREADS` is described in more detail [here](https://lancedb.github.io/lance/performance.html).
